tests/fw_rules_tests/policies/expected_output/label_expr_test_1-scheme_output.csv

```diff
@@ -4,22 +4,22 @@
 
 "query","src_ns","src_pods","dst_ns","dst_pods","connection",
 "test_app_label, config: test2","","","","","",
-"","","ip block: 0.0.0.0/0","[default]","[app in (A,B)]","All connections",
+"","","ip block: 0.0.0.0/0","[default]","[(app in (A,B))]","All connections",
 "","","ip block: 0.0.0.0/0","[default]","[c]","All connections",
 
 "query","src_ns","src_pods","dst_ns","dst_pods","connection",
 "test_app_label, config: test3","","","","","",
-"","","ip block: 0.0.0.0/0","[default]","[!has(app) or app in (dev)]","All connections",
+"","","ip block: 0.0.0.0/0","[default]","[!has(app) or (app in (dev))]","All connections",
 
 "query","src_ns","src_pods","dst_ns","dst_pods","connection",
 "test_app_label, config: test4","","","","","",
-"","","ip block: 0.0.0.0/0","[default]","[app in (dev)]","All connections",
+"","","ip block: 0.0.0.0/0","[default]","[(app in (dev))]","All connections",
 "","","ip block: 0.0.0.0/0","[default]","[d]","All connections",
 
 
 "query","src_ns","src_pods","dst_ns","dst_pods","connection",
 "test_tier_and_app_label, config: test5","","","","","",
-"","","ip block: 0.0.0.0/0","[default]","[{app in (G)} and {tier in (W)}]","All connections",
+"","","ip block: 0.0.0.0/0","[default]","[{(app in (G))} and {(tier in (W))}]","All connections",
```

Comment:
What are the curly braces for? Seems like we can do without them...

Reply:
updated implementation so that these extra parenthesis are only added when required (e.g. at label_expr_test_1-scheme_output)